edx_exams/apps/core/tests/test_api.py

```diff
@@ -289,6 +290,138 @@ def test_with_no_attempt(self):
         self.assertIsNone(get_attempt_by_id(111111111))
 
 
+@ddt.ddt
+class TestGetLatestAttemptForUser(ExamsAPITestCase):
+    """
+    Test for the API utility function `get_latest_attempt_for_user`
+    """
+    def setUp(self):
+        super().setUp()
+
+        self.course_id = 'course-v1:edx+test+f19'
+        self.content_id = '11111111'
+
+        self.exam = Exam.objects.create(
+            resource_id=str(uuid.uuid4()),
+            course_id=self.course_id,
+            provider=self.test_provider,
+            content_id=self.content_id,
+            exam_name='test_exam',
+            exam_type='proctored',
+            time_limit_mins=30,
+            due_date='2040-07-01 00:00:00',
+            hide_after_due=False,
+            is_active=True
+        )
+
+        self.user = User.objects.create(
+            id=3,
+            username="jerry",
+            email="jerry@example.com",
+            lms_user_id=2
+        )
+
+        # Create mock exam attempt data
```

Comment:
do we still need all of these created if you're setting up data in each test below?